uber/reports.py

```diff
@@ -0,0 +1,48 @@
+from uber.common import *
+
+
```

Comment:
One option to consider instead of "subclass ``ReportBase`` and override the ``write_row`` method" would be to allow plugins to set the write callback.  Like they could say

```PersonalizedBadgeReport(callback=lambda out, row: out.writerow(row if 'some_val' in row else []))```

or something along those lines.  That's mostly just a style preference, but in general "subclass this and override a method to provide custom behavior" is bigger in Java while "just pass a callback" is bigger in Python.